Classes/EventListener/TopwireRenderContentElementByContext.php

```diff
@@ -0,0 +1,140 @@
+<?php
+declare(strict_types=1);
+
+namespace Topwire\EventListener;
+
+use Topwire\ContentObject\TopwireContentObject;
+use Topwire\Context\TopwireContext;
+use TYPO3\CMS\Core\Attribute\AsEventListener;
+use TYPO3\CMS\Core\Information\Typo3Version;
+use TYPO3\CMS\Core\Routing\PageArguments;
+use TYPO3\CMS\Core\TypoScript\AST\Node\ChildNode;
+use TYPO3\CMS\Core\TypoScript\AST\Node\RootNode;
+use TYPO3\CMS\Core\Utility\MathUtility;
+use TYPO3\CMS\Frontend\Event\ModifyTypoScriptConfigEvent;
+
+#[AsEventListener(
+    identifier: 'TopwireRenderContentElementByContext',
+    event: ModifyTypoScriptConfigEvent::class,
+)]
+class TopwireRenderContentElementByContext
+{
+    private const turboStreamContentType = 'text/vnd.turbo-stream.html';
+
+    public function __invoke(ModifyTypoScriptConfigEvent $event): void
+    {
+        $request = $event->getRequest();
+
+        $context = $request->getAttribute('topwire');
+
+        if (!$context instanceof TopwireContext) {
+            return;
+        }
+
+        $routing = $request->getAttribute('routing');
+        if (!$routing instanceof PageArguments) {
+            return;
+        }
+
+        $pageType = $routing->getPageType();
+        if ($pageType === '') {
+            return;
+        }
+
+        $pageAstNode = $this->getPageByTypeNum($event->getSetupTree(), $pageType);
+        if ($pageAstNode === null) {
+            return;
+        }
+
+        // This is super dirty and hacky! It is not possible to modify the "setup" in this event
+        // but only the "config" part. But as the PHP AST is using objects and objects are passed by reference
+        // in php we can modify the setup part by using these references.
+        // So we can partially modify the setup part here. This is a hack and should be replaced by a proper
+        // event in the future once TYPO3 has a proper event for this.
+        $firstChildNode = null;
+        foreach ($pageAstNode->getNextChild() as $childNode) {
+            // We can't remove the node itself as we need to keep the object reference
+            // so we remove all children instead
+            foreach ($childNode->getNextChild() as $childChildNode) {
+                $childNode->removeChildByName($childChildNode->getName());
+            }
+
+            if (MathUtility::canBeInterpretedAsInteger($childNode->getName()) && $firstChildNode === null) {
+                // Keep the first content node as we modify it by reference
+                $firstChildNode = $childNode;
+                continue;
+            }
+
+            // Remove the content of the node to it does not do anything
+            $childNode->setValue('');
+            $childNode->updateName('');
+        }
```

Comment:
As already noted this is a dirty hack. And it is broken. I'll not work as expected due to TypoScript caching mechanism. I could not find a way to solve this for now. I'll just fix this now im my typoscript as i can't find a general solution without patching the core / requesting new events.